pkg/virt-config/config-map.go

```diff
@@ -67,56 +62,37 @@ const (
 
 type ConfigModifiedFn func()
 
-func getConfigMap() *k8sv1.ConfigMap {
-	virtClient, err := kubecli.GetKubevirtClient()
-	if err != nil {
-		panic(err)
-	}
-
-	var cfgMap *k8sv1.ConfigMap
-	err = wait.PollImmediate(time.Second*1, time.Second*10, func() (bool, error) {
-
-		namespace, curErr := clientutil.GetNamespace()
-		if err != nil {
-			return false, err
-		}
-
-		cfgMap, curErr = virtClient.CoreV1().ConfigMaps(namespace).Get(ConfigMapName, metav1.GetOptions{})
-
-		if curErr != nil {
-			if errors.IsNotFound(curErr) {
-				logger := log.DefaultLogger()
-				logger.Infof("%s ConfigMap does not exist. Using defaults.", ConfigMapName)
-				cfgMap = &k8sv1.ConfigMap{}
-				return true, nil
-			}
-			return false, curErr
-		}
-
-		return true, nil
-	})
-
-	if err != nil {
-		panic(err)
-	}
-
-	return cfgMap
-}
-
 // NewClusterConfig represents the `kubevirt-config` config map. It can be used to live-update
 // values if the config changes. The config update works like this:
 // 1. Check if the config exists. If it does not exist, return the default config
 // 2. Check if the config got updated. If so, try to parse and return it
 // 3. In case of errors or no updates (resource version stays the same), it returns the values from the last good config
-func NewClusterConfig(configMapInformer cache.SharedIndexInformer, crdInformer cache.SharedIndexInformer, namespace string) *ClusterConfig {
+func NewClusterConfig(configMapInformer cache.SharedIndexInformer,
+	crdInformer cache.SharedIndexInformer,
+	kubeVirtInformer cache.SharedIndexInformer,
+	namespace string) *ClusterConfig {
+
+	defaultConfig := defaultClusterConfig()
+
+	objects := kubeVirtInformer.GetStore().List()
+	var kubeVirtName string
+	for _, obj := range objects {
```

Comment:
One lst point: I would sleep better, if on the long run our logic can deal with unusual but not impossible behaviour on the CR level. One can for instance delete the operator, force-delete the CR, recreate the operator and create a CR with a different name. It should be good enough if we just always list all elements from the cache and choose the CR which has a status.

Reply:
Sorry, I'm not sure I understand. Do you mean this check should be done on every call to get the config to make sure we have the correct KV? So instead of getting it once during `NewClusterConfig`, instead check for the KV name on each run of `getConfigFromKubeVirtCR`? 

In the example it would mean the controller, api, and handler did not get deleted / restarted but he KV got renamed, if I am understanding the edge case correctly.

Reply:
Yes exactly.

Reply:
Wile absolutely an edge case I would not say there are no reasons to do such things at all. It would be great if we are as declarative and eventually consistent as possible. If we don't do this check, people would have to restart manually the components after such an undertaking.

Reply:
makes sense to me! I have updated it now :)